highfidelity/hifi
# support for input controller pose filters

Adds support for various filters to run against "pose" style inputs. This will be useful in implementing mo-cap from vive pucks without going through JS update loops.

Test plan:
- smoke test
- make sure hand controllers still work normally
- run [this script](https://raw.githubusercontent.com/ZappoMan/hifi/6daf68b338282ad68f615339bc185c48a38d64ce/script-archive/controllers/puppetFeet3.js) and verify that the avatars left foot will "puppet" around from your left hand. It should be approximately 1 meter below your hand, and rotated from your hand rotation, but otherwise track your hand. (note: this is not a QA fail, if the IK doesn't always move the foot to exactly this position).

TODO:
- verify that with the post transform filter, the velocity is correct
- implement a test that shows this "child of the tracker" style filter
